src/scenic/core/distributions.py

```diff
@@ -19,1186 +19,1189 @@
-		if valueType is None:
-			ty = type_support.underlyingType(obj)
-			valueType = self.inferType(ty, attribute)
-		super().__init__(obj, valueType=valueType)
-		self.attribute = attribute
-		self.object = obj
-
-	@staticmethod
-	def inferType(ty, attribute):
-		"""Attempt to infer the type of the given attribute."""
-		# If the object's type is known, see if we have an attribute type annotation.
-		try:
-			hints = typing.get_type_hints(ty)
-			attrTy = hints.get(attribute)
-			if attrTy:
-				return attrTy
-		except Exception:
-			pass	# couldn't get type annotations
-
-		# Handle unions
-		origin = typing.get_origin(ty)
-		if origin == typing.Union:
-			types = []
-			for option in typing.get_args(ty):
-				if (option is type(None) and not hasattr(None, attribute)):
-					# None does not have this attribute; accessing it will raise an
-					# exception, so we can ignore this case for type inference.
-					continue
-				res = AttributeDistribution.inferType(option, attribute)
-				types.append(res)
-			return type_support.unifierOfTypes(types) if types else object
-
-		# We can't tell what the attribute type is.
-		return object
-
-	def sampleGiven(self, value):
-		obj = value[self.object]
-		return getattr(obj, self.attribute)
-
-	def evaluateInner(self, context):
-		obj = valueInContext(self.object, context)
-		return AttributeDistribution(self.attribute, obj)
-
-	def supportInterval(self):
-		obj = self.object
-		if isinstance(obj, MultiplexerDistribution):
-			attrs = (getattr(opt, self.attribute) for opt in obj.options)
-			return unionOfSupports(supportInterval(attr) for attr in attrs)
-		return None, None
-
-	def __call__(self, *args):
-		vty = self.object._valueType
-		retTy = None
-		if vty is not object:
-			func = getattr(vty, self.attribute, None)
-			if func:
-				if isinstance(func, property):
-					func = func.fget
-				retTy = typing.get_type_hints(func).get('return')
-		return OperatorDistribution('__call__', self, args, valueType=retTy)
-
-	def __repr__(self):
-		return f'{self.object!r}.{self.attribute}'
+    """Distribution resulting from accessing an attribute of a distribution"""
+
+    _deterministic = True
+
+    def __init__(self, attribute, obj, valueType=None):
+        if valueType is None:
+            valueType = self.inferType(obj, attribute)
+        super().__init__(obj, valueType=valueType)
+        self.attribute = attribute
+        self.object = obj
+
+    @staticmethod
+    def inferType(obj, attribute):
+        """Attempt to infer the type of the given attribute."""
+        # If the object's type is known, see if we have an attribute type annotation.
+        ty = type_support.underlyingType(obj)
+        try:
+            hints = typing.get_type_hints(ty)
+            attrTy = hints.get(attribute)
+            if attrTy:
+                return attrTy
+        except Exception:
+            pass    # couldn't get type annotations
+
+        # Check for a @property defined on the class with a return type
```

Comment:
Something went wrong with the merge here: the case for `Union` is missing (it should come just above this).